### PR TITLE
Use remark for easily detectable markdown documentation

### PR DESCRIPTION
**Pull request by halkeye**
Original plan was to just support the wiki export, but was able to support a bunch more. I really just wanted to see if I could do it based on what I learned on the gatsby-jenkins.io project. Pull out some markdown reading/rendering from api to local. Did this by marking specific content as text/markdown and letting the gatsby-transform-remark handle converting it. There's tons of really cool remark plugins like the auto linking headers\.

Changes:

* Upgrade gatsby (cause didn't want to figure out what versions of remark worked with older version)
* Create Plugin Wiki objects that will be processed by markdown / remark  handler
* Link wiki objects to plugins
* Render wiki objects
* Syntax highlighting for code blocks
* Resize any youtube videos to fit the screen
* Autolink headers (ex http://next-cherries.surge.sh/scp/#version-19---)

Major wins.
* As there's no github API calls, its hitting the raw pages, it is much more reliable at getting the docs.
* Reduced plugin-site-api usage
* SO MANY GATSBY-REMARK PLUGINS - https://www.gatsbyjs.com/plugins/?=remark

Marking it as a draft as I kinda want to play around with it more, but open to feedback if its a good idea or not
cc @MarkEWaite @zbynek

### CODE REVIEW

**Comment by halkeye** [comment]
image
^^^ plugin site for /scp at time of writing

File: gatsby-config.js [hunk 1]
```diff
@@ -42,7 +42,48 @@ module.exports = {
...
+                },
+                {
+                    resolve: 'gatsby-remark-embed-youtube',
+                    options: {
+                        width: 800,
+                        height: 400
+                    }
+                },
+                {
+                    resolve: 'gatsby-remark-autolink-headers',
```
Comment (timja) [inline gatsby-config.js:76]: ❤️  excited for this

**Comment by MarkEWaite** [comment]
Strong +1 from me.  @stackscribe asked in Docs office hours last week about linking to locations inside a markdown page from other pages (like www.jenkins.io).  This looks like it will provide that.  Thanks for doing it!

**Comment by zbynek** [comment]
I'm sure there are a lot simpler solutions for "autolink headers" :) But this looks like a useful step towards simpler architecture. Any thoughts on AsciiDOC ([example](https://github.com/jenkinsci/git-plugin/blob/master/README.adoc))? Do you also want to parse it via some gatsby plugin?

**Comment by halkeye** [comment]
yea, the autolink headers that work on jenkins.io should work on plugins if we included it, but it seems the github HTML injects random HTML in there too.

https://www.gatsbyjs.com/plugins/gatsby-transformer-asciidoc/ does work, not fast or well, but does work. I'm using it on the jenkins.io gatsby project

**Comment by halkeye** [comment]
oh wow, I think this might be more important than I originally thought. https://plugins.jenkins.io/scp/ is failing 24 hours later. I'm guessing plugin-site-api just doesn't handle things.

**Comment by halkeye** [comment]
I've updated https://next-cherries.surge.sh with my latest build, so feel free to poke around and see if anything looks obviously broken.

**Comment by timja** [comment]
I can’t see any docs at least on my phone 

https://next-cherries.surge.sh/azure-credentials/

I tried slack plugin as well

**Comment by halkeye** [comment]
Thx I'll take a look tomorrow. I may have built it wrong locally. Gotta get
my demo site back up.

**Comment by halkeye** [comment]
Yea totally found the issue, was only rendering markdown, not (markdown or regular HTML) (plugin site api)

**Comment by timja** [comment]
Where's my heading links!
https://next-cherries.surge.sh/azure-credentials/

Content is there now

**Comment by halkeye** [comment]
> Where's my heading links!

sadly that wasn't the goal of this PR, the goal was to move some of the processing out of plugin-site-api, especially now how often I'm seeing failed documentation.

Next steps would be to move more of it into the system. (Handle ones that are specifically .adoc, then handle ones that are just root github, then see whats left over)

I'll do a more permanent fix for headers after this.

I'm leaning towards merging to fix how often I'm seeing docs failing to load

**Review by timja — APPROVED** [review]

**Pull request merged**